### PR TITLE
Change SuggestionCache to requests suggestions when there is no text in searchbox

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/JSUI-2004

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/misc/SuggestionsCache.ts [hunk 1]
```diff
@@ -2,10 +2,6 @@ export class SuggestionsCache<T> {
   private cache: { [hash: string]: Promise<T> } = {};
 
   getSuggestions(hash: string, suggestionsFetcher: () => Promise<T>): Promise<T | null> {
```
Comment (francoislg) [inline src/misc/SuggestionsCache.ts]: Should probably remove `| null` here and the same condition in `clearSuggestion`

File: test/misc/SuggestionsCacheTest.ts [hunk 2]
```diff
@@ -10,13 +10,6 @@ export function SuggestionsCacheTest() {
       cache = new SuggestionsCache<string>();
     });
 
-    it('should return null when the cache key is empty', async done => {
```
Comment (francoislg) [inline test/misc/SuggestionsCacheTest.ts]: Should probably add a test to validate that an empty string is working.
